protocol/apps.proto

```diff
@@ -67,3 +70,67 @@ message GetStatusResponse {
     // version info returned by the webhook
     string version_info = 3;
 }
+
+// As a performance optimization, we may wish to bundle together many key solicitation requests
+// and channel messages for the bot to respond to in a single webhook call. Thus, an isolated
+// group of channel messages and encryption keys, or a request for a key solicitation, is
+// represented here. For webhook calls regarding channel content, the payload may contain one or
+// more of these events.
+message EventPayload {
+    // A MessagesPayload represents a group of user messages in a channel that qualifies for the app
+    // to be notified. The included set of group encryption sessions in this message should have
+    // all the needed ciphertexts to decrypt the set of messages sent in the same payload.
+    message MessagesPayload {
+        repeated Envelope messages = 1;
+        repeated Envelope group_encryption_sessions_messages = 2;
+    }
+
+    // SolicitKeys represents one or more messages in the same channel that cannot be forwarded to the
+    // app because the app does not have keys available to decrypt it. In this case, the app must send
+    // a message to the stream to solicit keys for it's device for the session ids that cannot yet be
+    // decrypted.
+    message SolicitKeysPayload {
+        repeated string session_ids = 1;
+        bytes stream_id = 2;
+    }
```

Comment:
I don't think you need "Payload" on the nested messages, it becomes EventsPayload.MessagesPayload which is a mouthful